backend/danswer/server/models.py

```diff
@@ -26,7 +26,7 @@ class UserRoleResponse(BaseModel):
 
 class SearchDoc(BaseModel):
     semantic_identifier: str
-    link: str
+    link: str | None
```

Comment:
Why would a link be `None`?

Reply:
The logic is just propagated from the indexing flow. We wrote it with the assumption that sometimes we may not be able to get a link for all sources, which... may still be true, though we haven't seen any connectors like that so far